suite/tests/api/drdecode_aarch64.c

```diff
@@ -49,11 +57,46 @@ test_disasm(void)
         pc = disassemble_with_info(GD, pc, STDOUT, false /*no pc*/, true);
 }
 
+/* XXX: It would be nice to share some of this code w/ the other
+ * platforms but we'd need cross-platform register references or keep
+ * the encoded instr around and compare operands or sthg.
+ */
+static void
+test_noalloc(void)
+{
+    byte buf[128];
+    byte *pc, *end;
+    instr_t *instr;
+
+    instr = XINST_CREATE_load(GD, opnd_create_reg(DR_REG_X0),
+                              OPND_CREATE_MEMPTR(DR_REG_X1, 0));
+    end = instr_encode(GD, instr, buf);
+    ASSERT(end - buf < BUFFER_SIZE_ELEMENTS(buf));
+    instr_destroy(GD, instr);
+
+    instr_noalloc_t noalloc;
+    instr_noalloc_init(GD, &noalloc);
+    pc = decode(GD, buf, &noalloc.instr);
+    ASSERT(pc != NULL);
+    ASSERT(opnd_get_reg(instr_get_dst(&noalloc, 0)) == DR_REG_X0);
```

Comment:
I think the interchanging use of  &noalloc.instr and  &noalloc is going to confuse the user.

Reply:
See https://github.com/DynamoRIO/dynamorio/pull/4183#discussion_r392777844

Reply:
Resolved with the use of a separate instr_t var per the discussion below.